Add Linux support

pepibumur opened the pull request.

yonaskolb commented:
Interesting. Do we need to remove the auto generated equatable code? How is that affecting Linux? Would be shame to have to update those all by hand when making changes

yonaskolb commented:
Ah, I see because of NSObject and @ObjC. We could have a custom Sourcery template then

pepibumur commented:
I removed the autogenerated code and moved it to each individual object. I purposely removed the dependency with Sourcery to keep the project as lean as possible. Forgetting to run Sourcery and not adding new properties to the `isEqual` method result in the same thing so I'd rather not add the tool.

pepibumur commented:
The package is building in Linux and tests are running. The last part missing is is adding all the tests to the `XCTestManifest` file. If all the tests run successfully, I think we are good to merge this PR.

yonaskolb commented:
@pepibumur `swift test --generate-linuxmain`

pepibumur commented:
> @pepibumur swift test --generate-linuxmain

🤯 Danke!

kwridan commented on README.md:
````diff
@@ -108,6 +108,14 @@ object.
+```
+docker run --rm --volume "$(pwd):/package" --workdir "/package" swift:5.1.2 /bin/bash -c "swift package resolve && swift test --build-path ./.build/linux"
+```
+
````
👍 Having this documented makes it easy to test this out!

pepibumur commented:
This PR is stale and I don't foresee working on this in the near future. I'll close it. If anyone is interested in accomplishing this one, feel free to open a new PR.
